test-dladdr.c

```diff
@@ -28,6 +28,58 @@ int check_dladdr( const void *addr, const char *addrsym )
     return !result;
 }
 
+/**
+ * @brief return address from a symbol located in a shared lilbrary
+ * @param libname librray to get the address from
+ * @param addrsym symbol to get the address for
+ * @return pointer to symbol address or NULL
+ */
+void *get_symbol_address( const char *libname, const char *sym )
+{
+    void *library = NULL;
+    void *addr = NULL;
+
+    library = dlopen( libname, RTLD_GLOBAL );
+    if ( library == NULL )
+    {
+        fprintf( stderr, "could not open '%s'\n", libname );
+        return NULL;
+    }
+
+    addr = dlsym( library, sym );
+    dlclose( library );
+    if (!addr)
+        fprintf( stderr, "could not get address for symbol '%s'\n", sym );
+    return addr;
+}
+
+/**
+ * @brief check address from a symbol located in a shared lilbrary
+ * @param libname librray to get the address from
+ * @param addrsym symbol to get the address for
+ * @return 0 check passed
+ * @return 1 check failed
+ */
+int check_dladdr_by_dlopen( const char *libname, const char *sym )
+{
+    void *addr = get_symbol_address( libname, sym);
+    if (!addr)
+        return 1;
+    return check_dladdr( addr, sym );
+}
+
+#ifdef _WIN32
+#define HMODULE void*
+#define HANDLE void*
+#define LPCSTR char*
+#define DWORD long
+
+/* link to import thunk */
+HMODULE __attribute__((__stdcall__)) GetModuleHandleA (LPCSTR lpModuleName);
+/* link to directly to iat */
+__attribute__((dllimport)) HMODULE __attribute__((__stdcall__)) LoadLibraryExA (LPCSTR lpLibFileName, HANDLE hFile, DWORD dwFlags);
```

Comment:
Please use `__declspec` instead of `__attribute__`. `__declspec` is supported by MSVC, GCC and LLVM. `__attribute__` is not supported by MSVC compiler.